### PR TITLE
feat: add TableNodeCell component

### PR DESCRIPTION
**Pull request by stone-lyl**
Long strings/JSON should be truncated

| Before | After |
| ------ | ----- |
| ❌      | ✅    |
|image|image|

 New lines are not preserved (might be good default to keep table compact). Consider showing full/linebreaked content on cell/row click

https://github.com/ajthinking/data-story/assets/20497176/97276d9d-624a-4113-9134-f1955aee1374

### CODE REVIEW

File: packages/ui/src/components/Node/ItemCollection.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    const { headers, rows } = new ItemCollection(jsonData).toTable();
+
+    expect(headers).toEqual(expectedHeaders);
+    expect(rows).toEqual(expectedContent);
+  });
+
+  it('should correctly extract headers and rows from nested JSON data', () => {
+    const { headers, rows } = new ItemCollection([nested]).toTable();
+
+    expect(headers).toMatchInlineSnapshot(`
```
Comment (ajthinking) [inline packages/ui/src/components/Node/ItemCollection.test.ts:79]: Prefer toEqual per #165 I assume this is fixed once that merges 🙂

**Review by ajthinking — COMMENTED** [review]
Super happy about this one, thank you 🙏 

1. Is the video a previous one by mistake?

2. Somethings a bit off with border styling here:
image

3. Can you take a look at the "empty state" as well? It currently just highjacks the table header/rows which is a bit weird 😅 

image


4. How to copy; case short cell, case truncated cell? Via tooltop/on click? Please think about a solution for this.

**Review by ajthinking — COMMENTED** [review]
🟢 Preview mode:
image

🟡 Tooltip mode:
image

Can we make this pretty formatting (2 spaces)? That is: inline( truncated) just as it is now. When clicked, show a BIG linebreaked/indented JSON. I think I missed this one in requirements.

**Comment by ajthinking** [comment]
Render boolean bug?
image

**Comment by stone-lyl** [comment]
> Super happy about this one, thank you 🙏
> 
> 1. Is the video a previous one by mistake?
> 2. Somethings a bit off with border styling here:
> 
> image
> 3. Can you take a look at the "empty state" as well? It currently just highjacks the table header/rows which is a bit weird 😅
> 
> image
> 4. How to copy; case short cell, case truncated cell? Via tooltop/on click? Please think about a solution for this.

1. Yeah, I've updated the video. 😊
2. I've fixed the issue with table content overflow. 
3. Yeah, I set the header to initialize as an empty state. This way, the issue won't occur. 
4. Added a copy button (like the picture below ) on the tooltip. What do you think of this idea? 🤔

image

**Comment by ajthinking** [comment]
Changes looking good! Lets ship this 🥳⛴️

**Pull request merged**